File: from-3401-to-3500/3452-sum-of-good-numbers/solution.go
func sumOfGoodNumbers(nums []int, k int) int {
    result := 0
    n := len(nums)
    for i := 0; i < n; i ++ {
        if i - k >= 0 && nums[i - k] >= nums[i] || i + k < n && nums[i + k] >= nums[i] {
            continue
        }
        result += nums[i]
    }
    return result
}

File: from-3401-to-3500/3452-sum-of-good-numbers/solution.py
class Solution:
    def sumOfGoodNumbers(self, nums: List[int], k: int) -> int:
        n = len(nums)
        result = 0
        for i, num in enumerate(nums):
            if i - k >= 0 and nums[i - k] >= num:
                continue
            if i + k < n and nums[i + k] >= num:
                continue
            result += num

        return result